Tasks/FtpUploadV2/ftpuploadtask.ts

```diff
@@ -166,22 +166,22 @@ function getFtpOptions(): FtpOptions {
 
     if (tl.getInput("credsType") === "serviceEndpoint") {
         // server endpoint
-        const serverEndpoint: string = tl.getInput("serverEndpoint", true);
+        const serverEndpoint: string = tl.getInput("serverEndpoint", true)!;
         serverEndpointUrl = url.parse(
-            tl.getEndpointUrl(serverEndpoint, false)
+            tl.getEndpointUrl(serverEndpoint, false)!
         );
 
         const serverEndpointAuth: tl.EndpointAuthorization = tl.getEndpointAuthorization(
             serverEndpoint,
             false
-        );
+        )!;
```

Comment:
Why it can't be falsey value here?
![Uploading image.png…]()

Reply:
Next serverEndpointAuth is used to get username and password in the code. And the fields are marked as required in case of entering creds manually (not using of service connection). So IMO it's correct to require declaring of username and password here.